frontend/packages/git-service/src/services/github-service.ts

```diff
@@ -105,7 +112,7 @@ export class GithubService extends BaseService {
       const resp = await this.client.repos.getContents({
         owner: this.metadata.owner,
         repo: this.metadata.repoName,
-        path: 'Dockerfile',
+        path: `${this.metadata.contextDir}/Dockerfile`,
```

Comment:
I'm sure this works in almost all cases you tested. But because contextDir is defined as optional, I would like to request to check if contextDir is defined. So that we don't get an unexpected `undefined/Dockerfile` in some edge-cases. Similar you have done in `gitlab-service.ts`.

To archive this its maybe a good idea to create two helper functions `isFilePresent(filename: string)` and `getFileContent(filename: string)` which can do all these repeated worked in this file where `isDevfilePresent` ends as a redirect to just 

```tsx
  isDevfilePresent = () => isFilePresent('devfile.yaml')
  getDevfileContent = () => getFileContent('devfile.yaml')
```

Reply:
The `getRepoMetadata` function will make sure that `contextDir` will always be available, at least as an empty string, so IMO checking for that isn't needed. What do you think?
There are even test cases where `contextDir` isn't provided.

Updated with helper functions `isFilePresent` & `getFileContent`.